src/ActionMenu.tsx

```diff
@@ -71,23 +71,26 @@ const ActionMenuBase = ({
     [anchorContent, renderAnchor]
   )
 
-  const renderMenuItem: typeof Item = useCallback(
-    ({onAction: itemOnAction, ...itemProps}) => {
-      return renderItem({
-        ...itemProps,
+  const itemsToRender = useMemo(() => {
+    return items.map(item => {
+      if ('renderItem' in item) {
```

Comment:
Why are `ItemProps` with `renderItem` a special case? (i.e. Why is this early return here?) The `renderItem` function is called with other `ItemProps` as an argument, and it seems like we’d want to include e.g. `role: 'menuitem'` (merged in on line 82) to the `ItemProps` it receives.

Reply:
This is intended as a short-circuit for `ActionList.Divider` and any other `ItemInput` values that are not actually `ItemProps`.  Happy to change to another approach if you have a suggestion. As far as real `ItemProps` that supply a `renderItem`, I figured those already take full responsibility for how they are rendered, so should do the same with how actions are handled.

Reply:
After some discussion, we decided it is best to remove these short-circuits for now.  The `ActionList.Divider` does not actually spread any additional props, so there is no harm in passing them.  The danger in _not_ passing them is if we have a custom _item_ that expects things like `role: menuitem` to be passed in automatically.